services/clojars/clojars-release.tester.js

```diff
@@ -1,12 +1,15 @@
 'use strict'
 
+const Joi = require('@hapi/joi')
 const t = (module.exports = require('../tester').createServiceTester())
 
 t.create('clojars (valid)')
   .get('/prismic.json')
   .expectBadge({
     label: 'clojars',
-    message: /^\[prismic "([0-9][.]?)+"\]$/, // note: https://github.com/badges/shields/pull/431
+    message: Joi.string()
+      .regex(/^\[prismic "([0-9][.]?)+"\]$/)
+      .required(), // note: https://github.com/badges/shields/pull/431
```

Comment:
What would you think about pushing this functionality into `expectBadge`?

Reply:
In principle, we could do. Any ideas on how to check that `expected` is a regex given


```
> typeof /^[0-9]+$/
'object'
```

?

Reply:
Yea, `/asdf/ instanceof RegExp` is `true`.

Reply:
I've added the code in 66344c0 to allow `expected` to just be a regex and reverted 0b18d17 to restore the 3 occurrences where we were just passing a regex.

There are many more occurrences throughout the codebase where we're passing `message: Joi.string().regex(/foo/)` e.g:

https://github.com/badges/shields/blob/8621fe42d76b809e9ba4ae844920f8ed0373817a/services/jenkins/jenkins-plugin-version.tester.js#L22
https://github.com/badges/shields/blob/51d864efc106dbfc7bae4d4aa2d306517a18e323/services/discourse/discourse.tester.js#L95-L97
https://github.com/badges/shields/blob/76b2e2634a358606426607b39a24b5de4d98e266/services/github/github-downloads.tester.js#L48-L50
https://github.com/badges/shields/blob/6cbc7b587a532d70a0f3f13d1d61a93b2ff7387b/services/lgtm/lgtm-alerts.tester.js#L11
...and so on

or where we are acheiving the same thing using `withRegex()` e.g:

https://github.com/badges/shields/blob/1f29c22d3dd47794da72254d2b8527dc0232fea2/services/keybase/keybase-btc.tester.js#L10
https://github.com/badges/shields/blob/1f29c22d3dd47794da72254d2b8527dc0232fea2/services/spiget/spiget-latest-version.tester.js#L12
https://github.com/badges/shields/blob/83c3c709086f73f4ec1576a0121d188c21ce5982/services/puppetforge/puppetforge-module-endorsement.tester.js#L10
...and so on

We could just replace all of those with `message: /foo/` as there is no reason to wrap them in a Joi schema, but given it is not necessary to perform the upgrade to Joi 16, its probably better to do that in another PR if you prefer that style. There are quite a lot of them

Reply:
> its probably better to do that in another PR if you prefer that style

Agreed!